src/app.js

```diff
@@ -516,7 +516,13 @@ var run = function () {
     queryParams.update({ version: version });
     var url;
     if (version === 'builtin') {
-      url = 'soljson.js';
+      var location = window.document.location;
+      location = location.protocol + '//' + location.host + '/' + location.pathname;
```

Comment:
The case we're dealing with, the href is `http://ethereum.github.io/browser-solidity/#version=builtin` and we want an end result of `http://ethereum.github.io/browser-solidity/soljson.js`.